atdgen/src/omelange_emit.ml

```diff
@@ -577,6 +591,12 @@ and make_sum_writer ?type_annot (p : param)
                 ; Line (sprintf "%s %s" (encoder_ident "constr0")
                           (make_json_string json_cons))
                 ]
+            | Some v when open_enum ->
+                (* v should resolve to type string. *)
+                [ Line (sprintf "| %s%s x ->" tick ocaml_cons)
+                ; Inline (make_writer p v)
```

Comment:
Shouldn't this be inside a block? I'm taking the other impl as reference:

https://github.com/ahrefs/atd/blob/77a07ed1f184f370d95ff882fa3a31911eecdf0f/atdgen/src/oj_emit.ml#L436

Maybe would be helpful to review to have an overview on the things that changed from native impl?

Reply:
I don't know, that's what I used first, but the indentation feels worse:

```diff
diff --git a/atdgen/test/melange/melangespec_mel.expected.ml b/atdgen/test/melange/melangespec_mel.expected.ml
index 744d15f..35f4242 100644
--- a/atdgen/test/melange/melangespec_mel.expected.ml
+++ b/atdgen/test/melange/melangespec_mel.expected.ml
@@ -170,7 +170,7 @@ let write_with_open_enum = (
     | `Beta ->
     Atdgen_codec_runtime.Encode.constr0 "Beta"
     | `Other x ->
-    Atdgen_codec_runtime.Encode.string
+      Atdgen_codec_runtime.Encode.string
     x
   )
 )
@@ -205,7 +205,7 @@ let write_variant3 = (
     | B ->
     Atdgen_codec_runtime.Encode.constr0 "B"
     | A x ->
-    Atdgen_codec_runtime.Encode.string
+      Atdgen_codec_runtime.Encode.string
     x
   )
 )
```